pkg/replica/replica.go

```diff
@@ -1118,5 +1119,13 @@ func (r *Replica) GetRemainSnapshotCounts() int {
 }
 
 func (r *Replica) getDiskSize(disk string) int64 {
-	return util.GetFileActualSize(r.diskPath(disk))
+	ret := util.GetFileActualSize(r.diskPath(disk))
+	if ret == -1 {
+		errMessage := fmt.Sprintf("Fail to getfile %v size", r.diskPath(disk))
```

Comment:
On second thought, why this function doesn't report an error? Should it be more straightforward if this reports the error?

Reply:
Since it's called from [ListDisks func](https://github.com/longhorn/longhorn-engine/blob/b33374258c994d9da0808f2d09161c10a65e3de9/pkg/replica/replica.go#L1087) and the return for that is `map[string]DiskInfo` type, I don't find a way to return error there.

Reply:
We need to refactor ListDisks() to allow it to return error then.

Reply:
In fact, we need to be more careful here.

Is ListDisks() call in the critical path or not? For example, does the failure of the call will result in a data path error or not? That will determine if we need to error out or not. Otherwise, we might have some unintended consequences.

Reply:
I think this is from `cmd.LsReplicaCmd`, and it's not in data path. It's ok to return whatever we found, and report the error through the monitor ping. Please let me know if you think we need to refactor ListDisks(), and this will impact the whole call path.